test/rihanna/job_test.exs

```diff
@@ -266,6 +266,28 @@ defmodule Rihanna.JobTest do
     end
   end
 
+  describe "mark_reenqueued/3" do
+    test "retains the rihanna_internal_meta field and sets due_at", %{pg: pg} do
+      job = insert_job(pg, :retried, 2)
+
+      %{rows: [[true]]} =
+        Postgrex.query!(pg, "SELECT pg_try_advisory_lock(#{@class_id}, $1)", [job.id])
+
+      now =
+        DateTime.utc_now()
+        |> DateTime.to_unix()
+
+      due_at = DateTime.from_unix!(now + 30)
+
+      mark_reenqueued(pg, job.id, due_at)
+
+      updated_job = get_job_by_id(pg, job.id)
+
+      assert updated_job.due_at |> DateTime.truncate(:second) == due_at
+      assert updated_job.rihanna_internal_meta["attempts"] == 2
```

Comment:
Can we get a second re-enqueue and make sure this count goes to 3?

Reply:
That's the check to ensure that meta is maintained, not incremented. It matches up with `insert_job(pg, :retried, 2)`. I can try to make that more clear in the test.